src/Coordinate.java

```diff
@@ -15,7 +15,17 @@ public Coordinate() {
 	public String call() throws Exception {
 		if (!this.called) {
 			this.notifyObserver();
-			return "Miss";
+			this.called = true;
+			if (this.hasShip) {
+				Ship ship = getObservable();
+				if (ship.isSunk()) {
+					return "Sunk";
+				} else {
+					return "Hit";
+				}
+			} else {
+				return "Miss";
+			}
```

Comment:
This method is a little bloated, doing too many things at once, violating SOLID principles.